### PR TITLE
feat: explore, bookmark, single post page, fix: posts db structure

### PR DESCRIPTION
**Pull request by percius47**
features:
- Explore
- Bookmarks
- Single User Profile
- Single Post Page
fix: 
- Follow/Unfollow
- reaction icons color

[Live Site](https://frittr.netlify.app/)

### CODE REVIEW

File: fritter/src/backend/controllers/AuthController.js [hunk 1]
```diff
@@ -84,6 +85,7 @@ export const loginHandler = function (schema, request) {
         { _id: foundUser._id, username },
         process.env.REACT_APP_JWT_SECRET
       );
+      console.log("found user",foundUser.following);
```
Comment (msdeshmukh009) [inline fritter/src/backend/controllers/AuthController.js]: You forgot to remove consoles

File: fritter/src/features/user/components/EditProfileModal.jsx [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+          updateProfile({
+            editInput: {
+              ...currentUser,
+              ...editInput,
+              profileAvatar: data.url,
+            },
+            token,
+          })
+        );
+      })
```
Comment (msdeshmukh009) [inline fritter/src/features/user/components/EditProfileModal.jsx:52]: Here you can use `async/await` and `try{...}catch{...}`.

**Review by msdeshmukh009 — APPROVED** [review]
All the mentioned features are working as expected. 
Good to go.

File: fritter/src/features/user/components/FollowListModal.jsx [hunk 3]
```diff
@@ -29,10 +30,12 @@ export const FollowListModal = ({ followModal, setFollowModal }) => {
                 setFollowModal(false);
               }}
             >
+     
               <UserAvatar user={item} />
 
               <div className="flex flex-col -mt-0.5">
                 <span>{item.fullName}</span>
+                {console.log("followlis",item)}
                 <span className="text-sm text-lightGrey -mt-1">
```
Comment (gopalbbharadva) [inline fritter/src/features/user/components/FollowListModal.jsx]: remove this console.

File: fritter/src/features/user/components/ProfileDetails.jsx [hunk 4]
```diff
@@ -116,11 +117,14 @@ console.log("curr user",currentUser);
...
             className="hover:underline cursor-pointer"
             onClick={() =>
+              {
               setFollowModal(() => ({
                 show: true,
                 title: "Following",
                 list: following,
-              }))
+              }));
+              console.log("following",following);
```
Comment (gopalbbharadva) [inline fritter/src/features/user/components/ProfileDetails.jsx]: remove console.

**Review by gopalbbharadva — APPROVED** [review]

File: fritter/src/features/user/components/ProfileDetails.jsx (hunk 4 above)
Comment (percius47) [reply fritter/src/features/user/components/ProfileDetails.jsx]: thanks for reminding me this gopal

File: fritter/src/features/auth/authSlice.js [hunk 5]
```diff
@@ -10,7 +10,7 @@ export const loginHandler = createAsyncThunk(
 
     try {
       const { data, status } = await loginService(login.input);
-
+      console.log("login data",data);
```
Comment (GeetaPogul) [inline fritter/src/features/auth/authSlice.js:13]: remove console

**Pull request merged**